tests/by-util/test_date.rs

```diff
@@ -169,6 +169,14 @@ fn test_date_format_y() {
     scene.ucmd().arg("+%y").succeeds().stdout_matches(&re);
 }
 
+#[test]
+fn test_date_format_q() {
+    let scene = TestScenario::new(util_name!());
+
+    let re = Regex::new(r"^\d\n$").unwrap();
```

Comment:
I would restrict the allowed values a bit more:
```suggestion
    let re = Regex::new(r"^[1-4]\n$").unwrap();
```

Reply:
thanks! applied.